gmso/core/views.py

```diff
@@ -0,0 +1,143 @@
+from collections import defaultdict
+
+from gmso.core.angle import Angle
+from gmso.core.angle_type import AngleType
+from gmso.core.atom import Atom
+from gmso.core.atom_type import AtomType
+from gmso.core.bond import Bond
+from gmso.core.bond_type import BondType
+from gmso.core.dihedral import Dihedral
+from gmso.core.dihedral_type import DihedralType
+from gmso.core.improper import Improper
+from gmso.core.improper_type import ImproperType
+
+potential_attribute_map = {
+    Atom: "atom_type",
+    Bond: "bond_type",
+    Angle: "angle_type",
+    Dihedral: "dihedral_type",
+    Improper: "improper_type",
+}
+
+
+def get_identifier(potential):
+    """Get identifier for a topology potential."""
+    if isinstance(potential, AtomType):
+        return potential.name
+    if isinstance(potential, (BondType, AngleType, DihedralType, ImproperType)):
+        return potential.member_types or potential.member_classes
+
+
+def unique_potentials(potential_types):
+    """Filter unique potentials based on pre-defiend identifiers."""
+    visited = defaultdict(set)
+    for potential_type in potential_types:
+        identifier = get_identifier(potential_type)
+        if identifier not in visited[type(potential_type)]:
+            visited[type(potential_type)].add(identifier)
+
+            yield potential_type
+
+
+class PotentialFilters:
+    UNIQUE_NAME_CLASS = "unique_name_class"
+
+
+default_filters = {PotentialFilters.UNIQUE_NAME_CLASS: unique_potentials}
+
+
+class TopologyPotentialView:
+    """A potential view based on different filters for a topology's potentials.
+
+    Parameters
+    ----------
+    iterator: typing.Iterator, required=True
+        An iterator of either topology sites or connections from which to extract potentials from
+
+    Parameters
+    ----------
+    filter_by: str or func, default=None
+        If provided, filter the collected potentials by some function
+        see, default_filters for names of the default potential filters
```

Comment:
I think we should show a case of using a filter_by, just to be clear where you might import these default_filter from. People with less python familiarity won't be totally clear how this filter is being applied, so showing the basic one PotentialFilters.UNIQUE_NAME_CLASS is probably the best bet.

Reply:
I agree. But I think even better place to put these examples will be in the property decorator's docstring in topology class.